datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java

```diff
@@ -320,11 +320,12 @@ public DetachRecipeV4Response detachRecipe(SdxCluster sdxCluster, DetachRecipeV4
 
     public void updateRangerRazEnabled(SdxCluster sdxCluster) {
         String initiatorUserCrn = ThreadBasedUserCrnProvider.getUserCrn();
+        DetailedEnvironmentResponse environment = environmentClientService.getByName(sdxCluster.getEnvName());
```

Comment:
please use env crn instead of name, to avoid unnecessary db lookups

Reply:
also it would the best to call the getByCrn method with internal actor also since this is a service-to-service communication

Reply:
Updated